tests/nnstreamer_datarepo/unittest_datareposrc.cc

```diff
@@ -0,0 +1,782 @@
+          UNITTEST_STATECHANGE_TIMEOUT), 0);
+
+  gst_object_unref (pipeline);
+}
+
+/**
+ * @brief Test for reading a file with invalid param (start-sample-index)
+ * the number of total sample(mnist.data) is 10 (0~9)
+ */
+TEST (datareposrc, invalidStopSampleIndex1_n)
+{
+  GstElement *datareposrc = NULL;
+  int idx_out_of_range = -1;
+  int get_value;
+  gchar *file_path = NULL;
+  gchar *json_path = NULL;
+
+  file_path = get_file_path (filename);
+  json_path = get_file_path (json);
+
+  gchar *str_pipeline =
+      g_strdup_printf
+      ("gst-launch-1.0 datareposrc name=datareposrc location=%s json=%s "
+      "start-sample-index=0 epochs=2 tensors-sequence=0,1 !"
+      "other/tensors, format=static, num_tensors=2, framerate=0/1, "
+      "dimensions=1:1:784:1.1:1:10:1, types=float32.float32 ! fakesink",
+      file_path, json_path);
+  GstElement *pipeline = gst_parse_launch (str_pipeline, NULL);
+  g_free (str_pipeline);
+  g_free (file_path);
+  g_free (json_path);
+  ASSERT_NE (pipeline, nullptr);
+
+  datareposrc = gst_bin_get_by_name (GST_BIN (pipeline), "datareposrc");
+  ASSERT_NE (datareposrc, nullptr);
+
+  /* set invalid param */
+  g_object_set (GST_OBJECT (datareposrc), "stop-sample-index", idx_out_of_range,
+      NULL);
+  /** value "-1" of type 'gint' is invalid or out of range for property 'start-sample-index' of type 'gint'
+     default value is set */
+  g_object_get (GST_OBJECT (datareposrc), "stop-sample-index", &get_value,
+      NULL);
+  EXPECT_EQ (get_value, 0);
+
+  gst_object_unref (pipeline);
+}
+
+/**
+ * @brief Test for reading a file with invalid param (epochs)
+ */
+TEST (datareposrc, invalidEpochs0_n)
+{
+  GstElement *datareposrc = NULL;
+  int invalid_epochs = -1;
+  guint get_value;
+  gchar *file_path = NULL;
+  gchar *json_path = NULL;
+
+  file_path = get_file_path (filename);
+  json_path = get_file_path (json);
+
+  gchar *str_pipeline =
+      g_strdup_printf
+      ("gst-launch-1.0 datareposrc name=datareposrc location=%s json=%s "
+      "start-sample-index=0 stop-sample-index=9 tensors-sequence=0,1 ! "
+      "other/tensors, format=static, num_tensors=2, framerate=0/1, "
+      "dimensions=1:1:784:1.1:1:10:1, types=float32.float32 ! fakesink",
+      file_path, json_path);
+  GstElement *pipeline = gst_parse_launch (str_pipeline, NULL);
+  g_free (str_pipeline);
+  g_free (file_path);
+  g_free (json_path);
+  ASSERT_NE (pipeline, nullptr);
+
+  datareposrc = gst_bin_get_by_name (GST_BIN (pipeline), "datareposrc");
+  ASSERT_NE (datareposrc, nullptr);
+
+  /* set invalid param */
+  g_object_set (GST_OBJECT (datareposrc), "epochs", invalid_epochs, NULL);
+  /** value "-1" of type 'gint' is invalid or out of range for property 'start-sample-index' of type 'gint'
+     default value is set */
+  g_object_get (GST_OBJECT (datareposrc), "epochs", &get_value, NULL);
+  g_warning ("get_value: %d", get_value);
+  EXPECT_EQ (get_value, 1);
```

Comment:
`EXPECT_EQ (get_value, 1U)` to avoid build warning.

Reply:
Thank you.